tests/AddressBookControllerIntegrationTest.java

```diff
@@ -0,0 +1,291 @@
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.swing.table.DefaultTableModel;
+import org.junit.Rule;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.internal.stubbing.answers.DoesNothing;
+import org.mockito.junit.MockitoJUnitRunner;
+
+
+/**
+ * AddressBookControllerIntegrationTest.java
+ * This Test class contains integration tests for AddressBook Controller Class
+ * This class uses Mockito to mock classes for mock testing.
+ *
+ */
+
+@RunWith(MockitoJUnitRunner.class)
+class AddressBookControllerIntegrationTest {
+
+
+  Person person_mock; //Mock person class
+  //AddressBook test_AddressBook;
+  AddressBook addressBook_mock;// stub AddressBook class
+  AddressBook addressBook_spy;
+  AddressBookController controllerTest;
+
+
+  @BeforeEach
+  /**
+   * This method is ran before each test is run.
+   * Is intended to set up variables for test to use.
+   */
+  void setUp() {
+
+    //Mock Functionality of AddressBook
+    addressBook_mock = mock(AddressBook.class);
+    //Spy of AddressBook
+    addressBook_spy = spy(AddressBook.class);
+    //A Test Object of AddressBook
+    controllerTest = new AddressBookController(addressBook_mock);
+   //Mock of Person class
+    person_mock = mock(Person.class);
+
+  }
+
+  @AfterEach
+  /**
+   * This method runs after a test is done.
+   * Clears data for the next test ot use
+   */
+  void tearDown() {
```

Comment:
I still don't understand why the professor had code like this. It really serves absolutely no purpose to set to null, something that would become null right after anyway

Reply:
Agreed, waste of a method in my opinion.